contracts/src/LightClientV2.sol

```diff
@@ -82,31 +89,43 @@ contract LightClientV2 is LightClient {
         }
         // format validity check
         BN254.validateScalarField(newState.blockCommRoot);
+        BN254.validateScalarField(nextStakeTable.blsKeyComm);
+        BN254.validateScalarField(nextStakeTable.schnorrKeyComm);
+        BN254.validateScalarField(nextStakeTable.amountComm);
 
         // epoch-related checks
         uint64 lastUpdateEpoch = currentEpoch();
-        uint64 newEpoch = epochFromBlockNumber(newState.blockHeight, _blocksPerEpoch);
-        // advancing 1 epoch is only allowed if the epoch root (last block - 5) of the last epoch
-        // was submitted
-        // or if there is no last epoch (i.e. when current epoch is epoch 1)
-        require(newEpoch < lastUpdateEpoch + 2, MissingEpochRootUpdate());
-        if (
-            lastUpdateEpoch != 0
-                && epochFromBlockNumber(newState.blockHeight + 5, _blocksPerEpoch) != lastUpdateEpoch
-        ) {
-            require(isEpochRoot(finalizedState.blockHeight), MissingEpochRootUpdate());
+        uint64 newEpoch = epochFromBlockNumber(newState.blockHeight, blocksPerEpoch);
+        // after epoch gets activated, for each epoch, we allow updates from
+        // `start_block..=epoch_root`
+        // but disallow any transitioning blocks `epoch_block+1..=last_block`, effectively
+        // `epoch_root`
+        // is the "last block" of every epoch from light client's perspective.
+        require(!isGtEpochRoot(newState.blockHeight), MissingEpochRootUpdate());
```

Comment:
This check should be behind `newEpoch >= firstEpoch`

Reply:
yes you are right, (in a previous local version i had it, but later somehow revert it). 
 done in [0b95276](https://github.com/EspressoSystems/espresso-network/pull/2984/commits/0b9527670ede4325c2e3fdc361bba92ba7ad8e5f)